commerce-media-dl/src/main/java/com/liferay/commerce/media/dl/internal/DLCommerceMediaResolver.java

```diff
@@ -252,7 +281,57 @@ protected FileEntry getFileEntry(HttpServletRequest httpServletRequest)
 
 		long fileEntryId = GetterUtil.getLong(pathArray[pathArray.length - 2]);
 
-		return _dlAppLocalService.getFileEntry(fileEntryId);
+		return getFileEntry(fileEntryId);
+	}
+
+	protected FileEntry getFileEntry(long fileEntryId) {
+		try {
+			return _dlAppLocalService.getFileEntry(fileEntryId);
+		}
+		catch (PortalException pe) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(pe, pe);
+			}
+
+			return null;
+		}
+	}
+
+	protected void sendDefaultMediaBytes(
+			HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse, String contentDisposition,
+			long groupId)
+		throws IOException {
+
+		try {
+			CommerceMediaDefaultImageConfiguration
```

Comment:
@ethanbustad  Can we add a cache layer using portal-cache ?  So we avoid this logic all the request, as well for the other method